x-pack/plugins/infra/public/hooks/use_trial_state.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { API_BASE_PATH as LICENSE_MANAGEMENT_API_BASE_PATH } from '../../../license_management/common/constants';
+
+export enum TrialStatusLoadState {
```

Comment:
Small nit: As this is hitting an external API it's probably not as relevant. But with our other data fetching hooks we normally use the `useTrackedPromise` hook (https://github.com/elastic/kibana/blob/master/x-pack/plugins/infra/public/utils/use_tracked_promise.ts), `useHTTPRequest` also wraps `useTrackedPromise`, and this uses strings for states rather than enums. Do we want to match that here? Or at least use the same naming convention? (`pending`, `resolved` etc?).

Reply:
I can use `useTrackedPromise`. It makes sense